### PR TITLE
Feature/sphinx docs

### PR DESCRIPTION
**Pull request by nklapste**
Adding some sphinx documentation to the project.

Misc cleanup and improvement of graypy's documentation.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -76,14 +76,22 @@ jobs:
...
+        - sphinx-build docs/ build/
+      after_script: skip
     - stage: deploy
       deploy:
         python:
           - 3.6
         provider: pypi
         user: nklapste
         password:
-          secure: Ryu87EzCW+k/quo2yC05cejU29zo9Xwlv9f2zxwMhCDRwMGt4GCSJ4zR7GXvYT0B5H6S+Y+gCxTmSSA53V1aRvCX0A42i6P9jef2yWR34aYe0+lmYwwW1wNKo/SLD1x5cXw59m5pcVGPvAVY9aQ88DwR9gn6aMr76A6LJet/ykY=
```
Comment (nklapste) [inline .travis.yml]: Add back to travis file

File: graypy/handler.py [hunk 2]
```diff
@@ -379,36 +372,36 @@ def send(self, s):
 
 
 class GELFTCPHandler(BaseGELFHandler, SocketHandler):
-    """Graylog Extended Log Format TCP handler"""
+    """GELF TCP handler"""
 
     def __init__(self, host, port=12201, **kwargs):
         """Initialize the GELFTCPHandler
 
         :param host: The host of the Graylog server.
```
Comment (nklapste) [inline graypy/handler.py]: Maybe document as  "GELF TCP input host"? Using a similar format as to what is documented by the `GELFHTTPHandler`

File: graypy/handler.py [hunk 3]
```diff
@@ -47,49 +47,47 @@
...
         :type fqdn: bool
 
         :param localname: If ``fqdn`` is :obj:`False` and ``localname`` is
-            specified, used the specified hostname to populate the
+            specified, use the specified hostname to populate the
             ``host`` GELF field.
         :type localname: str or None
 
         :param facility: If specified, replace the ``facility`` GELF field
             with the specified value. Additionally, the LogRecord.name will
```
Comment (nklapste) [inline graypy/handler.py]: wrap LogRecord.name in literal backticks?

**Comment by severb** [comment]
I was under the impression that we already merged this and your other outstanding diffs. Should I take a casual glance at the changes and merge them?

**Comment by nklapste** [comment]
This is mainly a update and amendment of past docs. 

This also adds sphinx documentation generation. But, its actual build on readthedocs is untested, last time I checked graypy does indeed have a readthedocs endpoint, but, I have no access to such.

Any review would be welcome @severb! If any of the docs seem confusing or contrived please let me know.

**Comment by severb** [comment]
I added you as a maintainer on RTD. The changes LGTM, please feel free to push them whenever you feel like it.

**Pull request merged**